client/emv/cmdemv.c

```diff
@@ -462,26 +477,26 @@ int CmdHFEMVExec(const char *cmd) {
 	struct tlv *pdol_data_tlv = dol_process(tlvdb_get(tlvRoot, 0x9f38, NULL), tlvRoot, 0x83);
 	if (!pdol_data_tlv){
 		PrintAndLog("ERROR: can't create PDOL TLV.");
-		return 4;
+		dreturn(4);
 	}
 	
 	size_t pdol_data_tlv_data_len;
 	unsigned char *pdol_data_tlv_data = tlv_encode(pdol_data_tlv, &pdol_data_tlv_data_len);
 	if (!pdol_data_tlv_data) {
 		PrintAndLog("ERROR: can't create PDOL data.");
-		return 4;
+		dreturn(4);
 	}
 	PrintAndLog("PDOL data[%d]: %s", pdol_data_tlv_data_len, sprint_hex(pdol_data_tlv_data, pdol_data_tlv_data_len));
 
 	PrintAndLog("\n* GPO.");
 	res = EMVGPO(true, pdol_data_tlv_data, pdol_data_tlv_data_len, buf, sizeof(buf), &len, &sw, tlvRoot);
 	
 	free(pdol_data_tlv_data);
-	free(pdol_data_tlv);
+	//free(pdol_data_tlv); --- free on exit.
```

Comment:
But it is not freed by dreturn().   tlvdbfree required as well on dreturn()?  ???

Reply:
fixed. i plain to do some refactoring before making card scan